client/packages/host/src/components/SyncProgress/SyncProgress.tsx

```diff
@@ -10,127 +10,116 @@ import {
-            if (!isOperational && index == SyncSteps.Push) return null;
-            const isActiveAndError = !!error && active && !completed;
-
-            return (
-              <Step key={index} active={active} completed={completed}>
-                <StepLabel
-                  icon={
-                    isActiveAndError && (
-                      <AlertIcon sx={{ color: 'error.main' }} />
-                    )
-                  }
-                  error={isActiveAndError}
-                  optional={
-                    progress && (
-                      <Box
-                        display={'flex'}
-                        justifyContent="center"
-                      >{`${progress.done}/${progress.total}`}</Box>
-                    )
-                  }
-                >
-                  {label}
-                </StepLabel>
-              </Step>
-            );
-          }
-        )}
-      </Stepper>
+    <Box display="flex" flexDirection={'column'}>
+      <HorizontalStepper steps={steps} colour={colour} />
       {error && <ErrorWithDetails {...error} />}
     </Box>
   );
 };
 
-// This is the order of sync operations on server
-enum SyncSteps {
-  // Only in initialisation mode
-  PrepareInitial,
-  // Only in operational mode
-  Push,
-  PullCentral,
-  PullRemote,
-  Integrate,
-}
+const ProgressIndicator = ({
+  progress,
+  colour,
+}: {
+  progress?: Progress;
+  colour: StepperColour;
+}) => (
+  <Box
+    display={'flex'}
+    justifyContent="center"
+    fontSize={12}
+    color={`${colour}.light`}
+    whiteSpace="nowrap"
+  >
+    {progress ? `${progress.done} / ${progress.total}` : null}
+  </Box>
+);
+
+type Progress = {
+  total: number;
+  done: number;
+};
 
 const getSteps = (
   t: TypedTFunction<LocaleKey>,
-  syncStatus?: SyncStatus
-): {
-  label: string;
-  progress?: { total: number; done: number };
-  active: boolean;
-  completed: boolean;
-}[] => {
-  const getProgress = (progress?: SyncStatusWithProgressFragment | null) => {
-    if (!progress?.total) return;
-    const { total: total, done } = progress;
+  colour: StepperColour,
+  syncStatus?: SyncStatus,
+  isError?: boolean,
+  isOperational?: boolean
+): StepDefinition[] => {
+  const getProgress = (
+    progress?: SyncStatusWithProgressFragment | SyncStatusFragment | null
+  ) => {
+    if (progress?.__typename === 'SyncStatusNode' || !progress?.total) return;
+    const { total, done } = progress;
     return { total, done: done || 0 };
   };
 
-  const getStatus = (
+  const getStep = (
+    labelKey: LocaleKey,
     progress?: SyncStatusWithProgressFragment | SyncStatusFragment | null
-  ) => ({
-    active: !!progress?.started,
-    completed: !!progress?.finished,
-  });
+  ): StepDefinition => {
+    const completed = !!progress?.finished;
+    const active = !completed && !!progress?.started;
+    const isActiveAndError = isError && active && !completed;
+    const icon = isActiveAndError ? (
```

Comment:
I guess this is quite specific to SyncProgress, glad you kept it here rather then move to HorizontalStepper

Reply:
yeah, wasn't sure.. thought that the calling component could decide how to show the error. Still not totally sure.. but I thought that this was the safest for now